File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.py
def fibonacci(n):
    zero = [1, 0, 1]
    one = [0, 1, 1]
    if n >= 3:
        for i in range(2, n):
            zero.append(zero[i] + zero[i - 1])
            one.append(one[i] + one[i - 1])
    print(f'{zero[n]} {one[n]}')

t = int(input())
for i in range(t):
    n = int(input())
    fibonacci(n)

File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());

        for (int i = 0; i < t; i++) {
            fibonacci(Integer.parseInt(br.readLine()));
        }
    }

    private static void fibonacci(int n) {
        List<Integer> zero = new ArrayList<>(List.of(1, 0, 1));
        List<Integer> one = new ArrayList<>(List.of(0, 1, 1));

        if (n >= 3) {
            for (int i = 2; i < n; i++) {
                zero.add(zero.get(i) + zero.get(i - 1));
                one.add(one.get(i) + one.get(i - 1));
            }
        }

        StringBuilder sb = new StringBuilder();
        sb.append(zero.get(n));
        sb.append(' ');
        sb.append(one.get(n));

        System.out.println(sb.toString());
    }
}